File: UglyTrivia/WithTests/C#/GoldenMasterTest.cs
using System;
using System.IO;
using NUnit.Framework;

namespace UglyTrivia
{
  [TestFixture]
  public class GoldenMasterTest
  {
    private bool DISABLED = false;

    private const int NUMBER_OF_SEEDS = 2;
    private const bool GENERATE = false;

    #pragma warning disable 162 // disable the "code unreachable" warning
    [Test]
    public void golden_master_test()
    {
      if (DISABLED) return;

      if (GENERATE)
      {
        generate_golden_master();
      }
      else
      {
        verify_game_against_golden_master();
      }
    }
    #pragma warning restore 162

    private void generate_golden_master()
    {
      Console.WriteLine(">>>>");
      for (int seed = 0; seed < NUMBER_OF_SEEDS; seed++)
      {
        StringWriter stdOut;
        try
        {
          stdOut = StdOut.captureStdOut();
          GameRunner.Run(new Random(seed));
        }
        finally
        {
          StdOut.restoreStdOut();
        }

        writeToGoldenMaster(stdOut.ToString(), seed);
      }
      Console.WriteLine("<<<<");
    }

    private void verify_game_against_golden_master() 
    {
      for (int seed = 0; seed < NUMBER_OF_SEEDS; seed++)
      {
        try
        {
          StringWriter stdOut = StdOut.captureStdOut();
          string goldenMaster = readFromGoldenMaster(seed);
          GameRunner.Run(new Random(seed));

          Assert.That(stdOut.ToString(), Is.EqualTo(goldenMaster),
                 "golden master mismatch for seed " + seed);
        }
        finally
        {
          StdOut.restoreStdOut();
        }
      }
    }

    private void writeToGoldenMaster(String content, long seed)
    {
            Console.WriteLine("\"" +
                content.Replace("\n", "\\n") +
                "\",");
    }

    private String readFromGoldenMaster(long seed) 
    {
      return GoldenMasterData.data[seed];
    }

  } 
}

File: UglyTrivia/WithTests/Javascript/golden_master_test.js
'use strict';

let ugly_trivia = require('./ugly_trivia.js');
let captures = require('./captures.js');
let golden_master = require('./golden_master_data.js');

let GENERATE = false;
let NUMBER_OF_SEEDS = 2;

describe('Golden Master Test', function () {

  var stdout;
  var random;

  if(GENERATE)
  {
    it('generates the Golden Master', function () {

      console.log(">>>>>");
  
      for (var seed = 0; seed < NUMBER_OF_SEEDS; seed++) {
        stdout = new captures.stdout();
        random = new captures.random(seed);

        ugly_trivia.Run();

        random.restore();
        stdout.restore();

        writeToGoldenMaster(stdout.capturedOutput);
      }
      console.log("<<<<<");
    });
  }
  else
  {
    it('checks simulation against the Golden Master', function () {
  
      for (var seed = 0; seed < NUMBER_OF_SEEDS; seed++) {
        stdout = new captures.stdout();
        random = new captures.random(seed);

        ugly_trivia.Run();

        random.restore();
        stdout.restore();

        var goldenMaster = readFromGoldenMaster(seed);
        stdout.capturedOutput.should.equal(goldenMaster, 
          "Golden Master mismatch for seed " + seed);
      }
    });
  }

  function writeToGoldenMaster(content) {
     console.log("\"" +
            content.replace(/\n/g, "\\n") +
                "\",");
  }

  function readFromGoldenMaster(seed) {
    return golden_master.data[seed];
  }


});

